src/MongoDB.Driver.Core/Core/Operations/QueryHelper.cs

```diff
@@ -60,13 +60,28 @@ public static BsonDocument CreateReadPreferenceDocument(ServerType serverType, R
             }
 
             // simple ReadPreferences of Primary and SecondaryPreferred are encoded in the slaveOk bit
-            if (readPreference.ReadPreferenceMode == ReadPreferenceMode.Primary || readPreference.ReadPreferenceMode == ReadPreferenceMode.SecondaryPreferred)
+            //if (readPreference.ReadPreferenceMode == ReadPreferenceMode.Primary || readPreference.ReadPreferenceMode == ReadPreferenceMode.SecondaryPreferred)
+            //{
+            //    var hasTagSets = readPreference.TagSets != null && readPreference.TagSets.Count > 0;
+            //    if (!hasTagSets && !readPreference.MaxStaleness.HasValue && readPreference.Hedge == null)
+            //    {
+            //        return null;
+            //    }
+            //}
+
+            // simple ReadPreferences of Primary and SecondaryPreferred are encoded in the slaveOk bit
+            switch (readPreference.ReadPreferenceMode)
             {
-                var hasTagSets = readPreference.TagSets != null && readPreference.TagSets.Count > 0;
-                if (!hasTagSets && !readPreference.MaxStaleness.HasValue)
-                {
+                case ReadPreferenceMode.Primary:
```

Comment:
Returning null here is only safe because the ReadPreference constructor has ensured that mode Primary is not mixed with any other options.

Reply:
This corresponds with what the spec says:
>For mode 'primary', drivers MUST NOT set the slaveOK wire protocol flag and MUST NOT use $readPreference